src/algorithms/tracking/SecondaryVertexFinder.cc

```diff
@@ -0,0 +1,419 @@
+// SPDX-License-Identifier: LGPL-3.0-or-later
+// Copyright (C) Dongwi H. Dongwi (Bishoy)
+
+#include "SecondaryVertexFinder.h"
+
+#include <Acts/Definitions/Units.hpp>
+#include <Acts/EventData/GenericBoundTrackParameters.hpp>
+#include <Acts/MagneticField/MagneticFieldProvider.hpp>
+#include <Acts/Propagator/EigenStepper.hpp>
+#include <Acts/Propagator/Propagator.hpp>
+#include <Acts/Propagator/VoidNavigator.hpp>
+#include <Acts/Utilities/AnnealingUtility.hpp>
+#include <Acts/Utilities/Delegate.hpp>
+#include <Acts/Utilities/Result.hpp>
+#include <Acts/Utilities/detail/ContextType.hpp>
+#include <ActsExamples/EventData/Track.hpp>
+#include <Acts/Vertexing/TrackAtVertex.hpp>
+#include <ActsExamples/EventData/Trajectories.hpp>
+#include <algorithms/service.h>
+#include <edm4eic/Cov4f.h>
+#include <edm4eic/Track.h>
+#include <edm4eic/TrackParameters.h>
+#include <edm4eic/Trajectory.h>
+#include <edm4eic/unit_system.h>
+#include <edm4hep/Vector2f.h>
+#include <edm4hep/Vector4f.h>
+#include <fmt/core.h>
+#include <fmt/format.h>
+#include <podio/RelationRange.h>
+#include <Eigen/Core>
+#include <cmath>
+#include <iostream>
+#include <limits>
+#include <string>
+#include <memory>
+#include <tuple>
+#include <utility>
+
+#include "Acts/Vertexing/AdaptiveGridTrackDensity.hpp"
+#include "Acts/Utilities/Logger.hpp"
+#include "extensions/spdlog/SpdlogToActs.h"
+#include "algorithms/interfaces/ActsSvc.h"
+#include "extensions/spdlog/SpdlogFormatters.h"
+
+void eicrecon::SecondaryVertexFinder::init(std::shared_ptr<spdlog::logger> log) {
+  m_log            = log;
+  auto& serviceSvc = algorithms::ServiceSvc::instance();
+  m_geoSvc         = serviceSvc.service<algorithms::ActsSvc>("ActsSvc")->acts_geometry_provider();
+  m_BField =
+      std::dynamic_pointer_cast<const eicrecon::BField::DD4hepBField>(m_geoSvc->getFieldProvider());
+  m_fieldctx = eicrecon::BField::BFieldVariant(m_BField);
+}
+
+std::tuple<std::unique_ptr<edm4eic::VertexCollection>, std::unique_ptr<edm4eic::VertexCollection>>
+eicrecon::SecondaryVertexFinder::produce(
```

Comment:
Please use the function signature `void process(const Input&, const Output&) const`. In particular, this function must be void, just be const, and should not create its own collections.